packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -179,7 +190,7 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     },
   );
 
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
```

Comment:
I don't know whether it is good to do this but after selecting an item, options array would be just that item and array is changed only on open and you can see that change: one selected item shortly changes to full options list.

Reply:
so wait, why was this change needed?

it does look very noticeable that the list is getting filtered before closing

Reply:
I explained in comment why it is needed 😀 I don't see anything switching on closing, this change was made to fix the opening.

Reply:
I noticed that too, that after selection only 1 item is visible while closing..

Reply:
Ok, when I turned off dev tools, it is visible for me too

Reply:
Changed other `useEffect` and its dep list, should be fixed an issue when closing. But this `useLayoutEffect` is still needed otherwise when data is loaded it will show skeletons -> No options -> options list.